core/trino-main/src/main/java/io/trino/connector/system/SchemasAuthorization.java

```diff
@@ -0,0 +1,143 @@
+            .column("authorization", createUnboundedVarcharType())
+            .build();
+
+    private final Metadata metadata;
+    private final AccessControl accessControl;
+
+    @Inject
+    public SchemasAuthorization(Metadata metadata, AccessControl accessControl)
+    {
+        this.metadata = requireNonNull(metadata, "metadata is null");
+        this.accessControl = requireNonNull(accessControl, "accessControl is null");
+    }
+
+    @Override
+    public Distribution getDistribution()
+    {
+        return SINGLE_COORDINATOR;
+    }
+
+    @Override
+    public ConnectorTableMetadata getTableMetadata()
+    {
+        return SCHEMAS_AUTHORIZATION;
+    }
+
+    @Override
+    public RecordCursor cursor(ConnectorTransactionHandle transactionHandle, ConnectorSession connectorSession, TupleDomain<Integer> constraint)
+    {
+        Session session = ((FullConnectorSession) connectorSession).getSession();
+        InMemoryRecordSet.Builder table = InMemoryRecordSet.builder(SCHEMAS_AUTHORIZATION);
+        for (CatalogSchemaAuthorization catalogSchemaAuthorization : getSchemasAuthorization(session, constraint)) {
+            TrinoPrincipal trinoPrincipal = catalogSchemaAuthorization.schemaAuthorization().trinoPrincipal();
+            table.addRow(
+                    catalogSchemaAuthorization.catalog(),
+                    catalogSchemaAuthorization.schemaAuthorization().schemaName(),
+                    trinoPrincipal.getType().toString(),
+                    trinoPrincipal.getName());
+        }
+        return table.build().cursor();
+    }
+
+    private List<CatalogSchemaAuthorization> getSchemasAuthorization(Session session, TupleDomain<Integer> constraint)
+    {
+        try {
+            return doGetSchemasAuthorization(session, constraint);
+        }
+        catch (RuntimeException exception) {
+            throw handleListingException(exception, "schemas_authorization", "system");
+        }
+    }
+
+    private List<CatalogSchemaAuthorization> doGetSchemasAuthorization(Session session, TupleDomain<Integer> constraint)
+    {
+        Domain schemaDomain = constraint.getDomain(1, VARCHAR);
+        Set<CatalogSchemaName> availableSchemas = listAllAvailableSchemas(
+                session,
+                metadata,
+                accessControl,
+                constraint.getDomain(0, VARCHAR),
+                schemaDomain);
+        Optional<String> schemaName = tryGetSingleVarcharValue(schemaDomain);
+        Map<String, Set<CatalogSchemaName>> groupedByCatalog = availableSchemas.stream()
+                .collect(groupingBy(CatalogSchemaName::getCatalogName, Collectors.toSet()));
+        ImmutableList.Builder<CatalogSchemaAuthorization> result = ImmutableList.builder();
+        for (String catalog : groupedByCatalog.keySet()) {
+            Set<SchemaAuthorization> allSchemasAuthorization = metadata.getSchemasAuthorizationInfo(session, new QualifiedObjectPrefix(catalog, schemaName, Optional.empty()));
```

Comment:
`QualifiedObjectPrefix` does not fit here. `name` is not here and schema should be a set.